tests/test_merge_db_functions.py

```diff
@@ -0,0 +1,477 @@
+
+  key, vals = parse_jobline(data)
+
+  assert (key == '1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F')
+  assert (vals == {
+      'miopenConvolutionFwdAlgoImplicitGEMM':
+          'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+      'miopenConvolutionFwdAlgoWinograd':
+          'ConvBinWinogradRxSf2x3g1,0.03856,0,miopenConvolutionFwdAlgoWinograd,not used',
+      'miopenConvolutionFwdAlgoDirect':
+          'ConvOclDirectFwdGen,0.0536,0,miopenConvolutionFwdAlgoDirect,not used',
+      'miopenConvolutionFwdAlgoGEMM':
+          'GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used'
+  })
+
+  algo = 'miopenConvolutionFwdAlgoImplicitGEMM'
+  assert (
+      vals[algo] ==
+      'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used'
+  )
+
+
+def test_parse_text_fdb_name():
+  master_file = '/data/tests/gfx90a_68.HIP.fdb.txt'
+
+  arch, num_cu, final_file, copy_files = parse_text_fdb_name(master_file)
+  assert (arch == "gfx90a")
+  assert (num_cu == '68')
+  assert (final_file == "/data/tests/gfx90a_68.HIP.fdb.txt")
+  assert (copy_files == [])
+
+
+def test_parse_text_pdb_name():
+  master_file = '/data/test/gfx1030_36.cd.pdb.txt'
+
+  arch, num_cu, final_file, copy_files = parse_text_pdb_name(master_file)
+  assert (arch == "gfx1030")
+  assert (num_cu == 36)
+  assert (final_file == "/data/tests/gfx1030_36.cd.pdb.txt")
+  assert (copy_files == [])
+
+
+def test_load_master_list():
+  master_file = '/data/tests/gfx803_36.HIP.fdb.txt'
+  master_list = load_master_list(master_file)
+
+  list1 = list(master_list.keys())
+  values = list(master_list.values())
+
+  if master_list == {}:
+    assert False
+  else:
+    assert True
+  if master_list:
+    assert (list(master_list.keys())[0] ==
+            '1-1-1-4x4-512-4-4-128-0x0-1x1-1x1-0-NCHW-FP32-B')
+    assert (list(master_list.values())[0] == {
+        'miopenConvolutionBwdDataAlgoWinograd':
+            'ConvBinWinogradRxSf2x3,0.06365,0,miopenConvolutionBwdDataAlgoWinograd,<unused>',
+        'miopenConvolutionBwdDataAlgoGEMM':
+            'GemmBwdRest,4.0393,32768,rocBlas,<unused>'
+    })
+
+
+def test_best_solver():
+
+  FdbKey_Set = ({
+      'miopenConvolutionFwdAlgoImplicitGEMM':
+          'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm, 0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+      'miopenConvolutionFwdAlgoWinograd':
+          'ConvBinWinogradRxSf2x3g1,0.03856,0,miopenConvolutionFwdAlgoWinograd,not used',
+      'miopenConvolutionFwdAlgoGEMM':
+          'GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used'
+  })
+
+  solver, time = best_solver(FdbKey_Set)
+  if (len(solver) == 0):
+    assert (False)
+  else:
+    assert (True)
+
+  if (float(time)):
```

Comment:
could be: assert float(time)
Is this testing if time != 0 ? That appears to be the effect here.

Reply:
yes, as it is set to 0 in test input